src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy

```diff
@@ -1591,7 +1623,231 @@ class PriceFloorsFetchingSpec extends PriceFloorsBaseSpec {
+
+        and: "PBS request on response object status should be in progress"
+        assert getRequests(bidResponse)[GENERIC.value]?.first?.ext?.prebid?.floors?.fetchStatus == INPROGRESS
+
+        and: "PBS bidderRequest status should be in progress"
+        def bidderRequest = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequest.ext.prebid.floors.fetchStatus == [INPROGRESS]
+
+        and: "Alerts.general metrics shouldn't be populated"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert !metrics[ALERT_GENERAL]
+
+        where:
+        bidRequest << [BidRequest.getDefaultBidRequest(), getBidRequestWithFloors().tap { it.ext.prebid.floors = null }]
+    }
+
+    def "PBS should emit error in log and response when data is invalid and floors fetching disabled for account and enabled for request"() {
+        given: "Default BidRequest with empty floors.data"
+        def bidRequest = bidRequestWithFloors.tap {
+            ext.prebid.floors.enabled = true
+            ext.prebid.floors.data = null
+        }
+
+        and: "Flush metrics"
+        flushMetrics(floorsPbsService)
+
+        and: "Account with disabled fetching"
+        def account = getAccountWithEnabledFetch(bidRequest.accountId).tap {
+            config.auction.priceFloors.fetch.enabled = false
+        }
+        accountDao.save(account)
+
+        and: "Default bid response"
+        def response = BidResponse.getDefaultBidResponse(bidRequest)
+        bidder.setResponse(bidRequest.id, response)
+
+        when: "PBS processes auction request"
+        def bidResponse = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS should log a warning"
+        def message = 'Price floor rules data must be present'
+        assert bidResponse.ext?.warnings[PREBID]*.code == [999]
+        assert bidResponse.ext?.warnings[PREBID]*.message == [FETCHING_DISABLED_WARNING_MESSAGE(message)]
+
+        and: "PBS should not add errors"
+        assert !bidResponse.ext.errors
+
+        and: "PBS should log a errors"
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, FETCHING_DISABLED_ERROR_LOG(bidRequest, message))
+        assert floorsLogs.size() == 1
+
+        and: "PBS request on response object status should be not in progress"
+        assert getRequests(bidResponse)[GENERIC.value]?.first?.ext?.prebid?.floors?.fetchStatus == NONE
+
+        and: "PBS request status shouldn't be in progress"
+        def bidderRequest = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequest.ext.prebid.floors.fetchStatus == [NONE]
+
+        and: "Alerts.general metrics should be populated"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert metrics[ALERT_GENERAL] == 1
+    }
+
+    def "PBS shouldn't emit error in log and response when data is invalid and floors fetching disabled for account and #requestFloors for request"() {
```

Comment:
same here:
`"PBS shouldn't emit error in log and response when floor data is empty and floors fetching disabled for account and #requestFloors for request"`